crates/net/src/http/method.rs

```diff
@@ -0,0 +1,105 @@
+
+/// HTTP methods that can be used in a fetch request.
+/// The methods as defined by the [fetch spec](https://fetch.spec.whatwg.org/#methods).
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+pub enum Method {
+    /// The OPTIONS method represents a request for information about the communication options
+    Options,
+    /// The GET method means retrieve whatever information (in the form of an entity) is
+    /// identified by the Request-URI.
+    Get,
+    /// The POST method is used to request that the origin server accept the entity enclosed
+    /// in the request as a new subordinate of the resource identified by the Request-URI
+    /// in the Request-Line.
+    Post,
+    /// The PUT method requests that the enclosed entity be stored under the supplied Request-URI.
+    Put,
+    /// The DELETE method requests that the origin server delete the resource identified by the Request-URI.
+    Delete,
+    /// The HEAD method is identical to GET except that the server MUST NOT return a message-body in the response.
+    Head,
+    /// The PATCH method requests that a set of changes described in the
+    /// request entity be applied to the resource identified by the Request-URI.
+    Patch,
+}
+
+impl Method {
+    /// Returns a `Method` from the given bytes.
+    pub fn from_bytes(src: &[u8]) -> Option<Method> {
+        match src {
+            b"OPTIONS" => Some(Method::Options),
+            b"GET" => Some(Method::Get),
+            b"POST" => Some(Method::Post),
+            b"PUT" => Some(Method::Put),
+            b"DELETE" => Some(Method::Delete),
+            b"HEAD" => Some(Method::Head),
+            b"PATCH" => Some(Method::Patch),
+            _ => None,
+        }
+    }
+
+    /// The OPTIONS method represents a request for information about the communication options
+    pub const OPTIONS: Method = Method::Options;
+    /// The GET method means retrieve whatever information (in the form of an entity) is
+    /// identified by the Request-URI.
+    pub const GET: Method = Method::Get;
+    /// The POST method is used to request that the origin server accept the entity enclosed
+    /// in the request as a new subordinate of the resource identified by the Request-URI
+    /// in the Request-Line.
+    pub const POST: Method = Method::Post;
+    /// The PUT method requests that the enclosed entity be stored under the supplied Request-URI.
+    pub const PUT: Method = Method::Put;
+    /// The DELETE method requests that the origin server delete the resource identified by the Request-URI.
+    pub const DELETE: Method = Method::Delete;
+    /// The HEAD method is identical to GET except that the server MUST NOT return a message-body in the response.
+    pub const HEAD: Method = Method::Head;
+    /// The PATCH method requests that a set of changes described in the
+    /// request entity be applied to the resource identified by the Request-URI.
+    pub const PATCH: Method = Method::Patch;
```

Comment:
These aliases just pollute the API, imo. `Method::POST` has no need when `Method::Post` exists